packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -100,6 +110,24 @@ contract TimelockGuard is IGuard, ISemver {
         return scheduledTransactions[_safe][_txHash];
     }
 
+    /// @notice Returns the list of all scheduled but not cancelled or executed transactions for
+    /// for a given safe
+    /// @dev WARNING: This operation will copy the entire set of pending transactions to memory,
+    /// which can be quite expensive. This is designed only to be used by view accessors that are
+    /// queried without any gas fees. Developers should keep in mind that this function has an
+    /// unbounded cost, and using it as part of a state-changing function may render the function
+    /// uncallable if the set grows to a point where copying to memory consumes too much gas to fit
+    /// in a block.
```

Comment:
The alternative to this is some kind of pagination like in the Safe's [ModuleManager](https://github.com/safe-global/safe-smart-account/blob/186a21a74b327f17fc41217a927dea7064f74604/contracts/base/ModuleManager.sol#L114). 

I would rather keep it simple and assume that this will only run offchain. If we're very worried about someone reading this on chain, then we could add `require(msg.sender == address(0))` which we've [done elsewhere](https://github.com/ethereum-optimism/optimism/blob/jm/timelock-getPending/packages/contracts-bedrock/src/universal/Proxy.sol#L25-L28) to enable/enforce offchain usage.